internal/drivers/distributed_manager.go

```diff
@@ -0,0 +1,170 @@
+package drivers
+
+import (
+	"context"
+	"fmt"
+	"runtime/debug"
+	"time"
+
+	"github.com/Masterminds/squirrel"
+
+	"github.com/drone-runners/drone-runner-aws/store"
+	"github.com/drone-runners/drone-runner-aws/types"
+	"github.com/drone/runner-go/logger"
+
+	"github.com/sirupsen/logrus"
+)
+
+type DistributedManager struct {
+	Manager
+}
+
+func NewDistributedManager(manager *Manager) *DistributedManager {
+	return &DistributedManager{
+		*manager,
+	}
+}
+
+func (d *DistributedManager) BuildPools(ctx context.Context) error {
+	query := types.QueryParams{RunnerName: d.runnerName}
+	return d.forEach(ctx, d.GetTLSServerName(), &query, d.buildPoolWithMutex)
+}
+
+// This helps in cleaning the pools
+func (d *DistributedManager) CleanPools(ctx context.Context, destroyBusy, destroyFree bool) error {
+	var returnError error
+	query := types.QueryParams{RunnerName: d.runnerName}
+	for _, pool := range d.poolMap {
+		err := d.cleanPool(ctx, pool, &query, destroyBusy, destroyFree)
+		if err != nil {
+			returnError = err
+			logrus.Errorf("failed to clean pool %s with error: %s", pool.Name, err)
+		}
+	}
+
+	return returnError
+}
+
+func (d *DistributedManager) GetInstanceStore() store.InstanceStore {
+	return d.instanceStore
+}
+
+func (d *DistributedManager) GetStageOwnerStore() store.StageOwnerStore {
+	return d.stageOwnerStore
+}
+
+func (d *DistributedManager) GetTLSServerName() string {
+	// keep server name constant since any runner should be able to send request to LE
+	return "distributed-dlite"
+}
+
+func (d *DistributedManager) IsDistributed() bool {
+	return true
+}
+
+// Instance purger for distributed dlite
+// Delete all instances irrespective of runner name
+func (d *DistributedManager) StartInstancePurger(ctx context.Context, maxAgeBusy, maxAgeFree time.Duration) error {
+	const minMaxAge = 5 * time.Minute
+	if maxAgeBusy < minMaxAge || maxAgeFree < minMaxAge {
+		return fmt.Errorf("distributed dlite: minimum value of max age is %.2f minutes", minMaxAge.Minutes())
+	}
+	if maxAgeBusy > maxAgeFree {
+		return fmt.Errorf(
+			"distributed dlite: max age of used instances (set to %.2fmin) should be less than max age of free instances (set to %.2fmin)",
+			maxAgeBusy.Minutes(), maxAgeFree.Minutes())
+	}
+
+	if d.cleanupTimer != nil {
+		panic("distributed dlite: purger already started")
+	}
+
+	t := time.Duration(maxAgeBusy.Minutes() * 0.9 * float64(time.Minute))
+	d.cleanupTimer = time.NewTicker(t)
```

Comment:
Add `defer d.cleanupTimer.Stop()`. If it's not there in the existing manager, can we add it there as well?

Reply:
But wouldn't it stop the timer ?

Reply:
you are right, maybe move it inside the go func and defer stop there

Reply:
But even in the go routine if we stop the next tick won't happen no ? The purger ticker should remain running.

Reply:
Ohh I see where you mean

Reply:
Done